### ISSUE TITLE
Conflicting messages between SC2002 and SC2024. Or, no right way to use "sudo tee" pattern.

### ISSUE THREAD

**Issue by memotype** [koalaman/shellcheck#1401]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2002 and SC2024
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6
- [x] I read the issue's wiki page, e.g. https://github.com/koalaman/shellcheck/wiki/SC2086
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
sudo tee foo < bar  # SC2024
cat bar | sudo tee foo  # SC2002

```

#### Here's what shellcheck currently says:
For SC2024 it says "sudo doesn't affect redirects. Use sudo cat file | ..", but when you change it to use "cat file | ..." SC2002 comes up saying "Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead."

#### Here's what I wanted or expected to see:
I think the problem is that SC2024 assumes you're always reading from a file that requires sudo to read. I think the `sudo tee ... < ...` pattern is the correct pattern (as opposed to `cat ... | sudo tee`), I just think the logic for detecting SC2024 should be tweaked a bit. Maybe only report it for `>`, `>>` and `|` when the right-hand side of the pipe doesn't have sudo, and don't report it for `<`.

I know these are just warnings, and I know there's no way to 100% detect when sudo will be needed on both ends of the pipe/redirect, just thought I'd point out that when you follow the recommendation from SC2024 you are given a different warning about the useless cat (SC2002).

**Comment by koalaman** [comment]
The suggestion is not to use `cat file | ..` but to use `sudo cat file | ..` . 

When you have the `sudo` there, it does not produce a UUOC warning.

**Comment by memotype** [comment]
Yes, I know what it means. What I'm saying is, the sudo might not be needed for the cat if the user has read access to the file. Putting sudo in front of the cat when it's not needed seems like bad advice.

**Comment by memotype** [comment]
Especially since the "sudo tee ... < ..." should be fine.

**Comment by koalaman** [comment]
The Wiki does mention this as an exception:

>If you want to run a command as root but redirect as the normal user, you can ignore this message.
